docs/releases.md

```diff
@@ -9,28 +9,49 @@
 
 We might expand the six week release schedule in case of:
 - No new updates to be added
 - In case of a large community event that takes the core team availability away (e.g. IPFS Conf, Dev Meetings, IPFS Camp, etc)
 
 ## Release Flow
 
-`go-ipfs` releases come in 4 stages:
+`go-ipfs` releases come in 4 stages designed to gradually roll out changes and reduce the impact of any regressions that may have been introduced. If we need to merge non-trivial features/fixes during the process, we start over at stage 1.
 
-- **Release Stage 1 - Internal testing** - Test the Release against our testing infrastructure, including interoperability, integration, test lab, multiple runtimes and the apps we've built (WebUI, Desktop, NPM on IPFS, HTTP Client Libraries). The intent is to make this stage fully automated (and somewhat is already), until then, we manually check a list and ensure that all tests have been run
-- **Release Stage 2 - Invite _early testers_ to try it out** - Reach out to our _early testers_ (i.e. projects that have volunteered to support `go-ipfs` by using their own test infrastructure and tell us the results)
-- **Release Stage 3 - Announce to the broader community** - Communicate to the community that a new Release Candidate is ready and that everyone is welcome to test it with us
-- **Release Stage 4 - Complete the Release** - Finalize the release, start the next release.
+### Stage 1 - Internal Testing
 
-The Release Stages are not linked to Release Candidate numbers, in fact, there can be multiple release candidate per stages as we catch bugs and improve the release itself.
+Before this stage, we expect _all_ tests (interop, testlab, performance, etc.) to pass.
 
-<p align="center">
-  <a href="https://ipfs.io">
-    <img src="https://gateway.ipfs.io/ipfs/QmaFtLxoCAm5vFQ9AftKkhJwSAdDdF1jzV9DfzW6gbXqFL/Paper.Sketches.23.png" width="450" />
-  </a>
-</p>
+At this stage, we'll:
+
+1. Start a partial-rollout to our own infrastructure.
+2. Test against ipfs and ipfs-shipyard applications.
+
+**Goal(s):**
+
+1. Make sure we haven't introduced any obvious regressions.
+2. Test the release in an environment we can monitor and easily roll back (i.e., our own infra).
+
+### Stage 2 - Public Beta
+
+At this stage, we'll announce the impending release to the community and ask for beta testers.
+
+**Goal:** Test the release in as many non-production environments as possible. This is relatively low-risk but gives us a _breadth_ of testing internal testing can't.
+
+### Stage 3 - Soft Release
+
+At this stage, we consider the release to be "production ready" and ask will ask the community and our early testers to (partially) deploy the release to their production infrastructure.
```

Comment:
In https://github.com/ipfs/go-ipfs/pull/6496, I had a separate stages for asking early testers to test in production and asking the rest of the community. I removed that here to try to keep this from dragging on but can add it back.

Thoughts @daviddias, @lanzafame?